### ISSUE TITLE
Query Selector PROP_TYPE syntax breaks on square brackets

### ISSUE THREAD

**Issue by davidje13** [airbnb/enzyme#1019]
It is currently not possible to query for an element by property if the value contains a `]` character. For example, the following fails:

```
find('input[name="field[0]"]')
```

This is due to a bad regex in `MountedTraversal` (and possibly elsewhere): https://github.com/airbnb/enzyme/blob/master/src/MountedTraversal.js#L287

The following regex is closer to being correct:

```
/=("(?:\\?.)*?"|'(?:\\?.)*?'|(?:\\?.)*?)]/
```

(this will support quoting with single or double quotes, and allow simple backslash escaping in all cases. But worth noting that since `coercePropValue` doesn't have any concept of escape sequences, the backslash sequences won't work fully)

To just support quoting without backslashes, the following could be used:

```
/=(".*?"|'.*?'|.*?)]/
```

Better to use a proper parser, but this quick-fix will at least allow selectors to use the common convention of array-like indexing in form names.

**Comment by ljharb** [comment]
We have many standing issues to improve selector support; in general though, using strings is discouraged, and the `.findWhere` functional API will serve you much better.

PRs to improve selector support are welcome, but we've been inclined to keep things the same until we can replace the entire selector implementation wholesale with a well-tested one.

**Comment by ljharb** [comment]
Closed in #1086
